### PR TITLE
Bip39 word suggestion method

### LINKED ISSUE #1336: BIP39 word suggestion API
To be used in BIP39 mnemonic phrase entry, provide an API to suggest words based on the currently entered partial words.

Simple to access ABI, usable from Swift and Kotlin.

Examples:
 rob -> robot, robust
 ai -> aim, air, airport, aisle

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

To be used in BIP39 mnemonic phrase entry, method suggests words based on the currently entered partial words.
See kotlin&swift tests.
Added to HDWallet.
Fixes #1336.

## Testing instructions

Unit tests, `*Mnemonic*`.

## Types of changes

* New feature (non-breaking change which adds functionality) 

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: swift/Tests/HDWalletTests.swift [hunk 1]
```diff
@@ -420,4 +420,9 @@ class HDWalletTests: XCTestCase {
 
         XCTAssertEqual(address, "band1pe8xm2r46rmctsukuqu7gl900vzprfsp4sguc3")
     }
+
+    func testBip39Suggest() {
+        XCTAssertEqual(HDWallet.bip39Suggest(prefix: "air"), "air airport")
```
Comment (vikmeup) [inline swift/Tests/HDWalletTests.swift:425]: should we return as an array of words? `["air", "airdrop"]` ?

**Review by vikmeup — APPROVED** [review]
looks good, just one suggestion

**Review by hewigovens — COMMENTED** [review]
overall looks good, I'd like to have a `TWMnemonic` interface where we can move this method and `isValid` method from `TWHDWallet`

```c
TW_EXPORT_STATIC_METHOD
bool TWMnemonicIsValid(TWString *_Nonnull mnemonic);

TW_EXPORT_STATIC_METHOD
bool TWMnemonicSuggest(TWString *_Nonnull prefix);
```

File: swift/Tests/HDWalletTests.swift (hunk 1 above)
Comment (hewigovens) [reply swift/Tests/HDWalletTests.swift:425]: this can be done by adding a swift extension

Comment (hewigovens) [reply swift/Tests/HDWalletTests.swift:425]:
```swift
extension Mnemonic {
    func suggest() -> [String]
}
```

Comment (catenocrypt) [reply swift/Tests/HDWalletTests.swift:425]: My arguments for using space-separated string:
- simplifies a lot crossing C++/C/JNI interfaces
- it is easy to split to array on other side
- mnemonic phrase is also sent as space-separated string to wallet core

Comment (hewigovens) [reply swift/Tests/HDWalletTests.swift:425]: on top of this, we can even return all wrong words + index for a whole mnemonic

Comment (hewigovens) [reply swift/Tests/HDWalletTests.swift:425]: @catenocrypt let's use string in c/cpp and create swift/kotlin native type extensions

Comment (catenocrypt) [reply swift/Tests/HDWalletTests.swift:425]: type extension: I've got no experience with this, do we have already sg like this?

Comment (catenocrypt) [reply swift/Tests/HDWalletTests.swift:425]: > on top of this, we can even return all wrong words + index for a whole mnemonic

Let's keep some logic to the client as well ;)  These calls are in-process function calls, no delay like for a BE API

Comment (hewigovens) [reply swift/Tests/HDWalletTests.swift:425]: I will add that for you

Comment (catenocrypt) [reply swift/Tests/HDWalletTests.swift]: @hewig, it is also possible that all words are valid, but the phrase not (due to checksum).

Comment (catenocrypt) [reply swift/Tests/HDWalletTests.swift]: I have added Mnemonic.isWordValid(), so words can be checked one-by-one

Comment (hewigovens) [reply swift/Tests/HDWalletTests.swift]: that's possible, if last word is treated as wrong checksum

**Review by hewigovens — APPROVED** [review]

**Pull request merged**